package/MDAnalysis/analysis/gnm.py

```diff
@@ -337,12 +353,9 @@ def generate_kirchoff(self):
             for jcounter in neighbour_atoms:
                 if jcounter > icounter and _dsq(positions[icounter], positions[jcounter]) <= cutoffsq:
                     iresidue, jresidue = residue_index_map[icounter], residue_index_map[jcounter]
-                    if self.MassWeight:
-                        contact = 1.0 / (len(self.ca.residues[iresidue].atoms) * len(self.ca.residues[jresidue].atoms)) ** 0.5
```

Comment:
So assuming `self.ca` is an AtomGroup, 
 - `ca.residues` will [be performing a `np.unique` on all the residue indices in the group](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L1522),
 - `residues[iresidue]` is just slicing a numpy array
 - `residue.atoms` [will be looking up a single entry in a residue->atoms table](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L2556).

So nothing too offensively slow from what I can see, but you might be able to calculate them all ahead of time (as these memberships won't change) if you're looping over these things a lot.